src/irc.nim

```diff
@@ -590,6 +614,14 @@ proc newAsyncIrc*(address: string, port: Port = 6667.Port,
   result.status = SockIdle
   result.userList = initTable[string, UserList]()
 
+  when defined(ssl):
+    if ssl:
+      try:
+        sslContext.wrapSocket(result.sock)
+      except:
+        result.sock.close()
```

Comment:
I think ``defer`` would work better here, can we use it?

Reply:
I got this from httpclient. This is the constructor, why would we defer socket close here?

Reply:
oh sorry, I'm being stupid. Never mind :)